From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SuperKali <hello@superkali.me>
Date: Fri, 11 Sep 2026 07:32:46 +0200
Subject: arm: mediatek: add MT7986 distro boot environment

MT7986 has no board config header content, so boards fall back to whatever
bootcmd is stored in the environment. Provide the standard load addresses
and distro boot targets, scanning USB first, then NVMe, then eMMC.

Signed-off-by: SuperKali <hello@superkali.me>
---
 include/configs/mt7986.h | 40 ++++++++++
 1 file changed, 40 insertions(+)

diff --git a/include/configs/mt7986.h b/include/configs/mt7986.h
index 111111111111..222222222222 100644
--- a/include/configs/mt7986.h
+++ b/include/configs/mt7986.h
@@ -9,4 +9,44 @@
 #ifndef __MT7986_H
 #define __MT7986_H
 
+#include <linux/sizes.h>
+
+#if IS_ENABLED(CONFIG_CMD_USB)
+#define BOOT_TARGET_USB(func)	func(USB, usb, 0)
+#else
+#define BOOT_TARGET_USB(func)
+#endif
+
+#if IS_ENABLED(CONFIG_NVME)
+#define BOOT_TARGET_NVME(func)	func(NVME, nvme, 0)
+#else
+#define BOOT_TARGET_NVME(func)
+#endif
+
+#if IS_ENABLED(CONFIG_CMD_MMC)
+#define BOOT_TARGET_MMC(func)	func(MMC, mmc, 0)
+#else
+#define BOOT_TARGET_MMC(func)
+#endif
+
+#define BOOT_TARGET_DEVICES(func)	\
+	BOOT_TARGET_USB(func)		\
+	BOOT_TARGET_NVME(func)		\
+	BOOT_TARGET_MMC(func)
+
+#define ENV_MEM_LAYOUT_SETTINGS		\
+	"scriptaddr=0x50000000\0"	\
+	"script_size_f=0x2000\0"	\
+	"kernel_addr_r=0x52000000\0"	\
+	"fdt_addr_r=0x62000000\0"	\
+	"ramdisk_addr_r=0x62180000\0"
+
+#include <config_distro_bootcmd.h>
+
+#define CFG_EXTRA_ENV_SETTINGS		\
+	"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0"	\
+	ENV_MEM_LAYOUT_SETTINGS		\
+	BOOTENV				\
+	"bootcmd=run distro_bootcmd\0"
+
 #endif
-- 
Armbian

